src/_pytest/terminal.py

```diff
@@ -463,6 +469,46 @@ def write_sep(
         self.ensure_newline()
         self._tw.sep(sep, title, fullwidth, **markup)
 
+    def write_hrule_single(
+        self,
+        title: Optional[str] = None,
+        fullwidth: Optional[int] = None,
+        **markup: bool
+    ) -> None:
+        self.write_sep(
+            "\u2500" if self.config.option.unicode else "-", title, fullwidth, **markup
+        )
+
+    def write_hrule_double(
+        self,
+        title: Optional[str] = None,
+        fullwidth: Optional[int] = None,
+        **markup: bool
+    ) -> None:
+        self.write_sep(
+            "\u2550" if self.config.option.unicode else "=", title, fullwidth, **markup
+        )
+
+    def write_hrule_error(
+        self,
+        title: Optional[str] = None,
+        fullwidth: Optional[int] = None,
+        **markup: bool
+    ) -> None:
+        self.write_sep(
+            "\u2049" if self.config.option.unicode else "!", title, fullwidth, **markup
```

Comment:
Not sure about this character "⁉", I think this one can be kept as is.

Reply:
OK, I've updated the pull request to use "!" in both the ASCII and Unicode case.